### PR TITLE
✨ Company settings

### PR DESCRIPTION
**Pull request by drapisarda**
This allow to edit the company settings. 
The email and email validation fields are managed on firebase. The other data are saved on angolia. 

REFACTORS:
I create a sidebar to be used where needed
I edited the Form component to get an object to iterate on, to create the fields, manage the submit button and the submit action

LIBRARY:
I added `vuex-persistedstate` to having the data on the browser session and so having them persistent on refresh

### CODE REVIEW

File: src/components/Form.vue [hunk 1]
```diff
@@ -4,16 +4,14 @@
...
         :element="element"
         :data-field="element.id"
         v-model="element.value"
         :class="`form__element form__element--${element.type}`"
       />
-      <input-generator
-        :class="'form__element--submit'"
-        :element="submitButton"
-      />
+      <button @click="submitAction">action</button>
```
Comment (drapisarda) [inline src/components/Form.vue]: Why man? @volcanioo

Comment (volcanioo) [reply src/components/Form.vue]: it's a temporary solution to avoid console errors

File: src/components/Form.vue [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        v-for="element in elements"
+        :key="element.id"
+        :element="element"
+        :data-field="element.id"
+        :required="element.required"
+        v-model="element.value"
+        :class="`form__element form__element--${element.type}`"
+      />
+      <button>{{ submitLabel }}</button>
+    </form>
```
Comment (volcanioo) [inline src/components/Form.vue:16]: Dude, this is great! I will refactor `Case Detail` with your `Form` content soon. <3

File: src/components/InputGenerator.vue [hunk 3]
```diff
@@ -10,7 +10,7 @@
       v-if="element.type === 'select'"
       v-model="element.value"
       @change="handleInput"
-      :required="element.required"
+      :required="Boolean(element.required)"
```
Comment (volcanioo) [inline src/components/InputGenerator.vue:13]: smooooooth ;) thank you <3

MVP

File: src/views/AnonymousLogin.vue [hunk 4]
```diff
@@ -1,6 +1,7 @@
 <template>
   <div>
     <base-header></base-header>
+    <!-- TODO Remove and use the component -->
```
Comment (volcanioo) [inline src/views/AnonymousLogin.vue:4]: Thank you! I won't remove this one and will tackle it soon dude <3

**Review by volcanioo — APPROVED** [review]
at-your-service

BAMMMMM!!! I truly appreciate your contribution in this project <3 Thank you so much mate, and I also appreciate that you did more than this, with the creation of `Form` component, I will use this in other pages to reduce the code for sure. Great detail man! AS USUAL:))))))))))))))))))))

**Pull request merged**